src/utils/deploy.ts

```diff
@@ -6,10 +6,16 @@
  */
 
 import { Interfaces, Flags } from '@oclif/core';
-import { ConfigAggregator, SfdxPropertyKeys } from '@salesforce/core';
+import { ConfigAggregator, Global, SfdxPropertyKeys, TTLConfig } from '@salesforce/core';
 import { Duration } from '@salesforce/kit';
 import { Nullable } from '@salesforce/ts-types';
-import { ComponentSet, ComponentSetBuilder, DeployResult, MetadataApiDeploy } from '@salesforce/source-deploy-retrieve';
+import {
+  ComponentSet,
+  ComponentSetBuilder,
+  MetadataApiDeploy,
+  MetadataApiDeployStatus,
+} from '@salesforce/source-deploy-retrieve';
+import { JsonMap } from '@salesforce/ts-types';
 import { getPackageDirs, getSourceApiVersion } from './project';
 import { API, TestLevel, TestResults } from './types';
```

Comment:
suggestion: SDR has a lot of types defined an exported.

ex: MetadataApiDeployOptions

would there be some value in using that instead of redefining types (ex: TestLevel, RunTestResult)